File: app/src/main/java/com/abdessamad/orbyt/data/models/Habit.kt
package com.abdessamad.orbyt.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(tableName = "habits")
data class Habit(
    @PrimaryKey
    val id: UUID = UUID.randomUUID(),
    val taskId: UUID,
    val targetDays: Int,
    val colorHex: String,
    val createdAt: Date = Date()
)

@Entity(tableName = "habit_logs")
data class HabitLog(
    @PrimaryKey
    val id: UUID = UUID.randomUUID(),
    val habitId: UUID,
    val logDate: Date,
    val isDone: Boolean = false,
    val createdAt: Date = Date()
)

File: fastapi/app/models/habit.py
from sqlalchemy import Column, String, DateTime, Boolean, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from app.database import Base

class Habit(Base):
    __tablename__ = "habits"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    task_id = Column(UUID(as_uuid=True), ForeignKey("tasks.id"), nullable=False)
    target_days = Column(Integer, nullable=False)
    color_hex = Column(String, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

class HabitLog(Base):
    __tablename__ = "habit_logs"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    habit_id = Column(UUID(as_uuid=True), ForeignKey("habits.id"), nullable=False)
    log_date = Column(DateTime, nullable=False)
    is_done = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.utcnow)
